version/version.go

```diff
@@ -5,26 +5,45 @@
 package version
 
 import (
+	_ "embed"
 	"fmt"
+	"strings"
 
 	version "github.com/hashicorp/go-version"
 )
 
-// The main version number that is being run at the moment.
-var Version = "1.5.0"
+// rawVersion is the current version as a string, as read from the VERSION
+// file. This must be a valid semantic version.
+//
+//go:embed VERSION
+var rawVersion string
+
+// dev determines whether the -dev prerelease marker will
+// be included in version info. It is expected to be set to "no" using
+// linker flags when building binaries for release.
+var dev string = "yes"
 
-// A pre-release marker for the version. If this is "" (empty string)
-// then it means that it is a final release. Otherwise, this is a pre-release
-// such as "dev" (in development), "beta", "rc1", etc.
-var Prerelease = "dev"
+// The main version number that is being run at the moment, populated from the raw version.
+var Version string
 
-// SemVer is an instance of version.Version. This has the secondary
-// benefit of verifying during tests and init time that our version is a
-// proper semantic version, which should always be the case.
+// A pre-release marker for the version, populated using a combination of the raw version
+// and the dev flag.
+var Prerelease string
+
+// SemVer is an instance of version.Version representing the main version
+// without any prerelease information.
 var SemVer *version.Version
 
 func init() {
-	SemVer = version.Must(version.NewVersion(Version))
+	semVerFull := version.Must(version.NewVersion(strings.TrimSpace(rawVersion)))
+	SemVer = semVerFull.Core()
+	Version = SemVer.String()
+
+	if dev == "no" {
```

Comment:
If there is already a prelease set in VERSION (e.g `-alpha`) would the desired behavior for local builds be that this is dropped in favour of `-dev` or the other way around?

Reply:
Dropping any existing prerelease in favor of `-dev`! That's been our convention previously, so seemed just as well to stick with it for now.